### PR TITLE
Onboarding form

### PR DESCRIPTION
**Pull request by rishabhsaxena**
Show onboarding form post signup for `appsmith-ee` users

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

File: app/client/src/pages/Applications/index.tsx [hunk 1]
```diff
@@ -859,36 +861,56 @@ type ApplicationProps = {
...
+              search={{
+                placeholder: "Search for apps...",
+                queryFn: this.props.searchApplications,
+              }}
+            />
+            <ApplicationsSection searchKeyword={this.props.searchKeyword} />
+          </>
+        ) : (
+          <OnboardingForm />
+        )}
```
Comment (hetunandu) [inline app/client/src/pages/Applications/index.tsx:913]: I think this looks cleaner but you can ignore it if you like

```suggestion
        {this.state.showOnboardingForm ? (
           <OnboardingForm />
        ) : (
         <>
            <ProductUpdatesModal />
            <LeftPane />
            <SubHeader
              search={{
                placeholder: "Search for apps...",
                queryFn: this.props.searchApplications,
              }}
            />
            <ApplicationsSection searchKeyword={this.props.searchKeyword} />
          </>
        )}
```

**Review by hetunandu — APPROVED** [review]

File: app/client/src/pages/Applications/index.tsx (hunk 1 above)
Comment (rishabhsaxena) [reply app/client/src/pages/Applications/index.tsx]: yeah i'll update it does look cleaner

**Review by hetunandu — APPROVED** [review]

**Comment by rishabhsaxena** [comment]
@hetunandu merging this in since we did a minor update post the review and the last tests run, so we can have this in the next release

**Pull request merged**